### PR TITLE
fix e2e-test for ci

### PR DESCRIPTION
**Pull request by jcantrill**
Fix e2e tests for use with ci

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -83,7 +84,7 @@ deploy: deploy-setup deploy-image
 deploy-example: deploy
 	oc create -n $(NAMESPACE) -f hack/cr.yaml
 
-test-e2e: deploy-image
+test-e2e:
```
Comment (ewolinetz) [inline Makefile]: why is this removed from here? the e2e tests currently rely on an image to be produced

File: test/e2e/clusterlogging_test.go [hunk 2]
```diff
@@ -49,75 +47,13 @@ func TestClusterLogging(t *testing.T) {
...
 	// run subtests
 	t.Run("clusterlogging-group", func(t *testing.T) {
+
 		t.Run("Cluster with fluentd", ClusterLoggingClusterFluentd)
 		time.Sleep(time.Minute * 1) // wait for objects to be deleted/cleaned up
 		t.Run("Cluster with rsyslog", ClusterLoggingClusterRsyslog)
 	})
 }
 
-func createRequiredClusterRoleAndBinding(f *framework.Framework, ctx *framework.TestCtx) error {
```
Comment (ewolinetz) [inline test/e2e/clusterlogging_test.go]: this didn't break e2e tests? I would think that this would cause errors in the CLO since it wouldn't have the cluster permissions to do anything with oauthclients and priorityclasses...

File: Makefile (hunk 1 above)
Comment (jcantrill) [reply Makefile]: It does rely on the image to be produced but CI has a mechanism for us to use the one built by CI and not one we have to both build and deploy.  This is the reason for the introduction of [1].  I think it's reasonable to require people who are manually running this test to explicitly build and deploy their own image.  IMO we need to cater to CI which is supposed to make our lives easier.
[1] https://github.com/openshift/cluster-logging-operator/pull/88/files#diff-aa5c0c7202e9c241e0c57a1d903f9ec2R21

File: test/e2e/clusterlogging_test.go (hunk 2 above)
Comment (jcantrill) [reply test/e2e/clusterlogging_test.go]: The tests were failing and I couldn't understand why.  I thought these were just duplicates for resources in the manifest but I see there are some differences.  Reverting.

File: hack/buildconfig-template.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (jcantrill) [inline hack/buildconfig-template.yaml]: Removing in favor of https://github.com/openshift/cluster-logging-operator/pull/89/files#diff-4c7729766fb0e3362bb4ecdd4880efbeR1

File: Makefile (hunk 1 above)
Comment (ewolinetz) [reply Makefile]: Ack, can we update our README to describe this then?

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,6 @@
 CURPATH=$(PWD)
 TARGET_DIR=$(CURPATH)/_output
+KUBECONFIG?=$(HOME)/.kube/config
```
Comment (richm) [inline Makefile]: Is this necessary?  by default, if KUBECONFIG is not set, the oc command will look in $HOME/.kube/config

File: test/e2e/clusterlogging_test.go [hunk 5]
```diff
@@ -13,13 +13,13 @@ import (
...
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
-	rbac "k8s.io/api/rbac/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	rbac "k8s.io/api/rbac/v1beta1"
 )
 
 var (
 	retryInterval        = time.Second * 5
-	timeout              = time.Second * 60
+	timeout              = time.Minute * 2
```
Comment (ewolinetz) [inline test/e2e/clusterlogging_test.go]: just to keep it consistent can you make this 120 sec instead?

File: hack/test-e2e.sh [hunk 6]
```diff
@@ -39,9 +45,9 @@ oc adm policy add-scc-to-user privileged -z rsyslog -n openshift-logging
 # allows rsyslog to query k8s api for all namespace/pod info
 oc adm policy add-cluster-role-to-user cluster-reader -z rsyslog -n openshift-logging
 
-TEST_NAMESPACE=openshift-logging go test ./test/e2e/... \
+TEST_NAMESPACE=${NAMESPACE} go test ./test/e2e/... \
   -root=$(pwd) \
-  -kubeconfig=$HOME/.kube/config \
+  -kubeconfig=${KUBECONFIG} \
```
Comment (richm) [inline hack/test-e2e.sh]: I see (comment from above) - is it necessary to pass in `-kubeconfig`?  It will not look at the env. var. $KUBECONFIG first, then $HOME/.kube/config?

File: test/e2e/clusterlogging_test.go (hunk 5 above)
Comment (jcantrill) [reply test/e2e/clusterlogging_test.go]: ack

File: hack/test-e2e.sh (hunk 6 above)
Comment (jcantrill) [reply hack/test-e2e.sh]: I have no other answer then the e2e test failed unless I provided a path where it could be something other then $HOME/.kube/config.  My assumptions is maybe the CI templates setup the environment such that it writes the config to someplace other then $HOME.  Wouldn't this be valid for a node where you do not intend to support users

Comment (richm) [reply hack/test-e2e.sh]: ok - weird

File: Makefile (hunk 1 above)
Comment (ewolinetz) [reply Makefile]: That's really vague... by that I don't understand what I would set that variable to. I was meaning something to state that you will need to `make deploy-image` manually...

File: README.md [hunk 7]
```diff
@@ -28,81 +28,68 @@ deploying the operator assumes:
...
-```
-$ REMOTE_REGISTRY=true make deploy-image
-```
+|Target|Description|
+|------|-----------|
+|`deploy-example`|Deploys all resources and creates an instance of the `cluster-logging-operator`, creates a ClusterLogging custom resource named `example`, starts the Elasticsearch, Kibana, and Fluentd pods running.|
+|`deploy`|Deploys all resources and creates an instance of the `cluster-logging-operator`, but does not create the CR nor start the components running.|
+|`undeploy`|Removes all `cluster-logging` resources and `openshift-logging` project|
+|`deploy-setup`|Setup the OKD cluster to support cluster-logging without deploying an instance of the `cluster-logging-operator`|
+|`deploy-image`|If you already have a remote cluster, and you just want to rebuild and redeploy the image to the remote cluster.  Use `make image` instead if you only want to build the imagebut not push it.|
```
Comment (ewolinetz) [inline README.md]: "image but"

File: Makefile (hunk 1 above)
Comment (jcantrill) [reply Makefile]: @ewolinetz better? https://github.com/openshift/cluster-logging-operator/pull/88/files#diff-04c6e90faac2675aa89e2176d2eec7d8R93

Comment (ewolinetz) [reply Makefile]: works for me!
